README.md

```diff
@@ -302,21 +328,9 @@ Two dictionaries (`ike` and `ipsec`) will be added to the `vpn_connections` dict
 
 | Parameter                                 | Description                                                                           | Type        | Required | Default                 | Libreswan Equivalent    |
 |-------------------------------------------|---------------------------------------------------------------------------------------|:-----------:|:--------:|-------------------------|-------------------------|
-| ike                                       | Defines information specific to setting up the IKE protocol.                          | dict        | no       | -                       | -                       |
-| ike.enc_alg                               | Which encryption algorithm to use for IKE.                                            | str         | no       | vpn\_enc\_alg           | ike                     |
-| ike.enc_keysize                           | Size of encryption key to use for IKE.                                                | str         | no       | -                       | ike                     |
-| ike.auth_alg                              | Which authentication algorithm to use for IKE.                                        | str         | no       | vpn\_auth\_alg          | ike                     |
-| ike.dh_group                              | Which Diffie-Hellman group to use for IKE.                                            | str         | no       | -                       | ike                     |
 | ike.lifetime                              | How long keying channel of an IKE connection should last before being renegotiated.   | str         | no       | vpn\_lifetime           | ikelifetime             |
```

Comment:
Why leave `ike.lifetime` and `esp.lifetime`?

Reply:
They've been kept on the table with the list of future implementations, since I did not add them, because my use case does not require them. But I can add if you feel it's good to have.

Reply:
I mean - we should either keep the entire table, or remove the entire table - I'm not sure why we would keep just `ike.lifetime`

Reply:
I'll work on that.

Reply:
tl;dr - I don't think we really want to add support for `ike` and `esp` at this time, unless there is a compelling use case.  The guidance I received from Paul Wouters (libreswan developer) at the time was that we should just use the platform defaults for these.

Sorry, it's been a while since I was deep into this - it's coming back to me - so in libreswan the parameter `ike` https://libreswan.org/man/ipsec.conf.5.html is a string like `ike=3des-md5;modp1024,aes-sha1;modp1536` - within this string is encoded several fields such as encryption algorithm, keysize, auth algorithm, dh_group, etc.  I thin the intention of the future was to allow the user to provide these explicitly rather than tying them to the proprietary libreswan string format.  If we now support the proprietary libreswan string format, we should use some other field like `ike_SOMETHING` so that we 1) can still use `ike` in the future 2) indicate that the value is the string encoding in the libreswan format (maybe `ike_libreswan` but that isn't very graceful)
@ueno feel free to chime in here

Reply:
I pushed a change to implement `ikelifetime` and `salifetime`.

I had to implement the `ike` and `esp` parameters because the default ones used by `libreswan` would not allow me to establish a connection with a NetApp filer under normal circumstances. I didn't try to change the encryption algorithms on the filer side, but I would argue that it's easier for the Linux admin side to adapt to those requirements, rather than the storage admin having to figure out what is and what is not supported by the client side. I hope this helps clarify my actions.

As to implementing this in a more granular fashion, while I understand your point, I have to kindly disagree with it. Breaking it down would only make sense if the role would need to support a different IPSec implementation, other than Libreswan, which would probably require a re-write of many other options anyway. But, if that's the direction the product wants to go, I'm open to it, and I can rename the option to whatever you feel more appropriate (or break it down in smaller chunks).

Reply:
ok - as we discussed privately - let's not worry about other ipsec implementations/providers for now - let's go with your original PR which used `ike` and `esp` as the string values used by libreswan.